docs/concepts/storage/volumes.md

```diff
@@ -972,50 +971,98 @@ specification, and to select the type of media to use, for clusters that have
+without adding them to the Kubernetes repository. 
 
-For storage vendors looking to create an out-of-tree volume plugin, [please refer to this FAQ](https://github.com/kubernetes/community/blob/master/sig-storage/volume-plugin-faq.md) for choosing between the plugin options.
+Before the introduction of `CSI` and `FlexVolume`, all volume plugins (like
+volume types listed above) were "in-tree" meaning they were built, linked,
+compiled, and shipped with the core Kubernetes binaries and extend the core
+Kubernetes API. This meant that adding a new storage system to Kubernetes (a
+volume plugin) required checking code into the core Kubernetes code repository.
+
+Both `CSI` and `FlexVolume` allow volume plugins to be developed independent of
+the Kubernetes code base, and deployed (installed) on Kubernetes clusters as
+extensions.
+
+For storage vendors looking to create an out-of-tree volume plugin, please refer
+to [this FAQ](https://github.com/kubernetes/community/blob/master/sig-storage/volume-plugin-faq.md).
 
 ### CSI
 
-CSI stands for [Container Storage Interface](https://github.com/container-storage-interface/spec/blob/master/spec.md),
-a specification attempting to establish an industry standard interface that
-container orchestration systems can use to expose arbitrary storage systems
-to their container workloads.
-Please read
-[CSI design proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md) for further information.
+{% assign for_k8s_version="v1.10" %}{% include feature-state-beta.md %}
+
+[Container Storage Interface](https://github.com/container-storage-interface/spec/blob/master/spec.md) (CSI)
+defines a standard interface for container orchestration systems (like
+Kubernetes) to expose arbitrary storage systems to their container workloads.
 
-<!-- TODO: add link to the kubernetes extension section -->
-The `csi` volume type is an in-tree CSI volume plugin for Pods to interact
-with external CSI volume drivers running on the same node.
-After having deployed a CSI compatible volume driver, users can use `csi` as the
-volume type to mount the storage provided by the driver.
+Please read the [CSI design proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md) for more information.
 
-CSI persistent volume support is an alpha feature in Kubernetes v1.9 and requires a
-cluster administrator to enable it. To enable CSI persistent volume support, the
-cluster administrator adds `CSIPersistentVolume=true` to the `--feature-gates` flag
-for apiserver, controller-manager, and kubelet.
+CSI support was introduced as alpha in Kubernetes v1.9 and moved to beta in
+Kubernets v1.10.
+
+Once a CSI compatible volume driver is deployed on a Kubernetes cluster, users
+may use the `csi` volume type to attach, mount, etc. the volumes exposed by the
```

Comment:
This means users are still supposed to use `csi` as the volume type when using volumes provisioned by these out-of-tree plugins ...

Reply:
Correct.